internal/store/clusterrolebinding.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2022 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	"context"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"k8s.io/kube-state-metrics/v2/pkg/metric"
+	generator "k8s.io/kube-state-metrics/v2/pkg/metric_generator"
+)
+
+var (
+	descClusterRoleBindingAnnotationsName     = "kube_clusterrolebinding_annotations"
+	descClusterRoleBindingAnnotationsHelp     = "Kubernetes annotations converted to Prometheus labels."
+	descClusterRoleBindingLabelsName          = "kube_clusterrolebinding_labels"
+	descClusterRoleBindingLabelsHelp          = "Kubernetes labels converted to Prometheus labels."
+	descClusterRoleBindingLabelsDefaultLabels = []string{"clusterrolebinding"}
+)
+
+func clusterRoleBindingMetricFamilies(allowAnnotationsList, allowLabelsList []string) []generator.FamilyGenerator {
+	return []generator.FamilyGenerator{
+		*generator.NewFamilyGenerator(
+			descClusterRoleBindingAnnotationsName,
+			descClusterRoleBindingAnnotationsHelp,
+			metric.Gauge,
+			"",
+			wrapClusterRoleBindingFunc(func(r *rbacv1.ClusterRoleBinding) *metric.Family {
+				annotationKeys, annotationValues := createPrometheusLabelKeysValues("annotation", r.Annotations, allowAnnotationsList)
+				return &metric.Family{
+					Metrics: []*metric.Metric{
+						{
+							LabelKeys:   annotationKeys,
+							LabelValues: annotationValues,
+							Value:       1,
+						},
+					},
+				}
+			}),
+		),
+		*generator.NewFamilyGenerator(
+			descClusterRoleBindingLabelsName,
+			descClusterRoleBindingLabelsHelp,
+			metric.Gauge,
+			"",
+			wrapClusterRoleBindingFunc(func(r *rbacv1.ClusterRoleBinding) *metric.Family {
+				labelKeys, labelValues := createPrometheusLabelKeysValues("label", r.Labels, allowLabelsList)
+				return &metric.Family{
+					Metrics: []*metric.Metric{
+						{
+							LabelKeys:   labelKeys,
+							LabelValues: labelValues,
+							Value:       1,
+						},
+					},
+				}
+			}),
+		),
+		*generator.NewFamilyGenerator(
+			"kube_clusterrolebinding_info",
+			"Information about clusterrolebinding.",
+			metric.Gauge,
+			"",
+			wrapClusterRoleBindingFunc(func(r *rbacv1.ClusterRoleBinding) *metric.Family {
+				labelKeys := []string{"roleref_kind", "roleref_name"}
```

Comment:
I assume e2e is failing due to this?

Reply:
I had a typo in the document. It should be passing now.